sopel/builtins/version.py

```diff
@@ -19,17 +19,55 @@
 GIT_DIR = os.path.join(PROJECT_DIR, '.git')
 
 
-def git_info():
-    head = os.path.join(GIT_DIR, 'HEAD')
+def _read_commit(gitdir: str, head: str) -> str | None:
+    """
+    Given paths to ``.git/`` and ``HEAD``, determine the associated commit hash
+    """
```

Comment:
```suggestion
    """Determine the associated commit hash from the git dir and its HEAD"""
```

If you want, you can detail each parameter with `:param gitdir: <desc>` stanza in the docstring's body.